### DISCUSSION TITLE
get value of counting

### DISCUSSION THREAD

**Discussion by najwabelarbi2** [roboflow/supervision#295] [Q&A] [answered]
Hello,
how can i get the value of the counter as a variable?

**Comment by SkalskiP** [comment]
Hi, @najwabelarbi2 👋🏻! Could you be a bit more specific?

**Reply by najwabelarbi2** [reply]
yes sure, 
I am using YOLOV8 and Roboflow Supervision to count objects that pass through a conveyor (in real time.
I used this tutorial to do so: https://www.youtube.com/watch?v=Mi9iHFd0_Bo&t=30s&ab_channel=Roboflow
Here is my code:

```python

from flask import Flask, render_template, Response
import cv2
from ultralytics import YOLO
import supervision as sv
from supervision import process_video
import numpy as np
from cv2 import VideoWriter
from cv2 import VideoWriter_fourcc
import matplotlib.pyplot as plt
import matplotlib.animation as animation
from flask import send_from_directory
import os

LINE_START = sv.Point(320, 0)
LINE_END = sv.Point(320, 480)

app = Flask(__name__)

@app.route('/')
def index():
    """Video streaming home page."""
    return render_template('index.html')

def generate_frames():  # generate frame by frame from camera
    line_counter = sv.LineZone(start=LINE_START, end=LINE_END)
    line_annotator = sv.LineZoneAnnotator(thickness=2, text_thickness=1, text_scale=0.5)
    box_annotator = sv.BoxAnnotator(
        thickness=2,
        text_thickness=1,
        text_scale=0.5
    )
    video = cv2.VideoWriter('yolov8_1.avi', VideoWriter_fourcc(*'MP42'), 25.0, (640, 480))
    # Load a model
    model = YOLO("yolov8n.pt") 

    for result in model.track(source=2, stream=True, save=False, half=True, agnostic_nms=True, tracker="bytetrack.yaml"):
        frame = result.orig_img
        detections = sv.Detections.from_yolov8(result)

        if result.boxes.id is not None:
            detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)

        labels = [
            f"{tracker_id} {model.model.names[class_id]} {confidence:0.2f}"
            for _, confidence, class_id, tracker_id in detections
        ]

        frame = box_annotator.annotate(
            scene=frame,
            detections=detections,
            labels=labels
        )

        line_counter.trigger(detections=detections)
        line_annotator.annotate(frame=frame, line_counter=line_counter)

        # Convert frames to byte strings
        ret, buffer = cv2.imencode('.jpg', frame)
        frame_bytes = buffer.tobytes()

        # Yield frames as a response to the SSE connection
        yield (b'--frame\r\n'
               b'Content-Type: image/jpeg\r\n\r\n' + frame_bytes + b'\r\n\r\n')

        
@app.route('/video_feed')
def video_feed():
    return Response(generate_frames(), content_type='multipart/x-mixed-replace; boundary=frame')

if __name__ == '__main__':
    app.run(debug=True)


```
So as you can see, I am trying to display the video feed  with the counter on a web page.
Now I want to get the value of the counter (The IN value) and store it on a csv file in real time.

/line_counter.py  found the in_count and out_count parameters the the trigger function ( in supervision/supervision/detection/line_counter.py,  but i don't know how to use it in my code to write the in count values in a csv file.

Do you have any idea how I can do that? 

Thank you in advance for your answer

**Reply by onuralpszr** [reply]
@najwabelarbi2  I did the changes for better readability for the question and code, it will help us the understand the problem better. Please keep it the mind for the future as well. 

Thank you.

**Reply by najwabelarbi2** [reply]
thank you for that, I will keep it in mind for the future

**Comment by SkalskiP** [comment, marked as answer]
Thanks @onuralpszr 🙏🏻! @najwabelarbi2, you should be able to access count values with `line_counter.in_count` and `line_counter.out_count`.

**Reply by najwabelarbi2** [reply]
Thnak you for your answer! also i have another question: what if I want to only display on screen the in values and remove out values? Like so:
![image](https://github.com/roboflow/supervision/assets/129163419/08f6ffa9-a39c-409d-8ee0-28faecb5cce7)

**Reply by onuralpszr** [reply]
@najwabelarbi2  

https://github.com/roboflow/supervision/blob/9c53bc16ac0440e87ace2dfcd0ac1f4af116b588/supervision/detection/line_counter.py#L84

```python
        custom_in_text: Optional[str] = None,
        custom_out_text: Optional[str] = None,
```
        
 That means modify this and add custom parts like this
        
```python      
line_annotator = sv.LineZoneAnnotator(thickness=2, text_thickness=1, text_scale=0.5, custom_in_text="custom_in_text",custom_out_text="custom_out_text")
```

**Reply by SkalskiP** [reply]
As for hiding in or out labels, you can do it now. But I'm happy to merge PR with that option added into 'LineZoneAnnotator` if you want to make it. 

Also, keep in mind that we are going to completely change `LineZone` and `LineZoneAnnotator` completely. And it might happen quite soon. @hardikdava already started to work on it:
- https://github.com/roboflow/supervision/issues/264
- https://github.com/roboflow/supervision/pull/266
